server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyBodyListener.scala

```diff
@@ -1,14 +1,14 @@
 package sttp.tapir.server.netty.internal
 
-import io.netty.buffer.ByteBuf
 import sttp.monad.MonadError
 import sttp.monad.syntax._
 import sttp.tapir.server.interpreter.BodyListener
+import sttp.tapir.server.netty.NettyResponse
 
 import scala.util.{Success, Try}
 
-class NettyBodyListener[F[_]](implicit m: MonadError[F]) extends BodyListener[F, ByteBuf] {
-  override def onComplete(body: ByteBuf)(cb: Try[Unit] => F[Unit]): F[ByteBuf] = {
+class NettyBodyListener[F[_]](implicit m: MonadError[F]) extends BodyListener[F, NettyResponse] {
+  override def onComplete(body: NettyResponse)(cb: Try[Unit] => F[Unit]): F[NettyResponse] = {
```

Comment:
in case of a `Right`, can we run `cb` only once the response is sent? (after the last chunk is processed)

Reply:
Maybe the following should work:
* we change `NettyReponse` to be a function `ChannelHandlerContext => (ChannelPromise, Either[...])` (probably at this point we need a dedicated type :) )
* in `NettyToResponseBody`, we use `ctx` to create the new promise
* here we return a function which calls the original & subscribes to the promise
* in `NettyServerHandler`, we pass in the ctx to create the response, and pass the created promise to the write function